kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/spec/runners/InstancePerTestSpecRunner.kt

```diff
@@ -105,15 +106,30 @@ internal class InstancePerTestSpecRunner(
     */
    private suspend fun executeInCleanSpec(test: TestCase): Result<Spec> {
       return createInstance(test.spec::class)
-         .flatMap { spec ->
-            runCatching {
-               extensions.intercept(spec) {
-                  extensions.beforeSpec(spec)
-                     .flatMap { run(it, test) }
-                     .flatMap { extensions.afterSpec(it) }
-               } ?: error("Failed to initialize spec ${test.spec::class.simpleName}")
-            }.flatten()
-         }
+         .flatMap { executeInGivenSpec(test, it) }
+   }
+
+   private suspend fun executeInGivenSpec(test: TestCase, spec: Spec): Result<Spec> {
+      return runCatching {
+         extensions.intercept(spec) {
+            extensions.beforeSpec(spec)
+               .flatMap { run(it, test) }
+               .flatMap { extensions.afterSpec(it) }
+         } ?: error("Failed to initialize spec ${test.spec::class.simpleName}")
+      }.flatten()
+   }
+
+   /**
+    * The first time we run a root test, we can use the already instantiated spec as the instance.
+    * This avoids creating specs that do nothing other than scheduling tests for other specs to run in.
+    * Eg, see https://github.com/kotest/kotest/issues/3490
+    */
+   private suspend fun executeInCleanSpecIfRequired(test: TestCase, defaultSpec: Spec): Result<Spec> {
+      return if (defaultInstanceUsed.compareAndSet(false, true)) {
+         Result.success(defaultSpec).flatMap { executeInGivenSpec(test, it) }
+      } else {
+         createInstance(test.spec::class).flatMap { executeInGivenSpec(test, it) }
```

Comment:
```suggestion
         executeInCleanSpec(test)
```

Duplicating the code of `executeInCleanSpec` here, I think?